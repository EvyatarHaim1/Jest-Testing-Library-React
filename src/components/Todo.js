import React from 'react';

function Todo() {
    return <div data-testid="todo-1">Hello world!</div>
}

export default Todo;
