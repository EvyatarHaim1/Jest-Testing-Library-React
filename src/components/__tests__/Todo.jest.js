import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Todo from '../Todo';

afterEach(() => {
    cleanup();
})

test('Should render non completed todo', () => {
    const todo = { id: 1, title: "wash dishes", completed: false };

    render(<Todo todo={todo} />);
    const todoElement = screen.getByTestId("todo-1");
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent("wash dishes");
    expect(todoElement).not.toContainHTML("<strike>");
});

test('Should render completed todo', () => {
    const todo = { id: 2, title: "make dinner", completed: true };

    render(<Todo todo={todo} />);
    const todoElement = screen.getByTestId("todo-2");
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent("make dinner");
    expect(todoElement).toMatchSnapshot("<strike>");
});

test("matches snapshot", () => {
    const todo = { id: 1, title: "wash dishes", completed: false };
    const tree = renderer.create(<Todo todo={todo} />).toJSON();
    expect(tree).toMatchSnapshot();
});