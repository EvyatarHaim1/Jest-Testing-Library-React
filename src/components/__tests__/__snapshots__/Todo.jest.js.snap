// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render completed todo: <strike> 1`] = `
<div
  data-testid="todo-2"
>
  <strike>
    <h1>
      make dinner
    </h1>
  </strike>
</div>
`;

exports[`matches snapshot 1`] = `
<div
  data-testid="todo-1"
>
  <h1>
    wash dishes
  </h1>
</div>
`;
